src/PointwiseFunctions/AnalyticSolutions/Tags.hpp

```diff
@@ -80,18 +73,231 @@ struct Error : db::PrefixTag, db::SimpleTag {
 /// \see ::Tags::AnalyticSolutions
 struct AnalyticSolutionsBase : db::BaseTag {};
 
-/// The analytic solutions for all `FieldTags`
+namespace detail {
+template <typename Tag>
+struct AnalyticImpl : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+}  // namespace detail
+
 template <typename FieldTags>
-struct AnalyticSolutions : AnalyticSolutionsBase, db::SimpleTag {
-  using type = ::Variables<db::wrap_tags_in<Analytic, FieldTags>>;
+struct AnalyticSolutions : ::Tags::AnalyticSolutionsBase, db::SimpleTag {
+  using field_tags = FieldTags;
+  using type = std::optional<
+      ::Variables<db::wrap_tags_in<detail::AnalyticImpl, FieldTags>>>;
+};
+
+template <typename Tag>
+struct is_analytic_solutions : std::false_type {};
+
+/// \cond
+template <typename FieldTagsList>
+struct is_analytic_solutions<AnalyticSolutions<FieldTagsList>>
+    : std::true_type {};
+/// \endcond
+
+template <typename Tag>
+struct Error : db::PrefixTag, db::SimpleTag {
+  using type = std::optional<typename Tag::type>;
+  using tag = Tag;
 };
 
-/// The analytic solutions for all `FieldTags`, or `std::nullopt` if no analytic
-/// solutions are available
+namespace detail {
+template <typename Tag>
+struct ErrorImpl : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+}  // namespace detail
+
 template <typename FieldTags>
-struct AnalyticSolutionsOptional : AnalyticSolutionsBase, db::SimpleTag {
-  static std::string name() { return "AnalyticSolutions"; }
-  using type =
-      std::optional<::Variables<db::wrap_tags_in<Analytic, FieldTags>>>;
+struct Errors : db::SimpleTag {
+  using field_tags = FieldTags;
+  using type = std::optional<
+      ::Variables<db::wrap_tags_in<detail::ErrorImpl, FieldTags>>>;
 };
+
+template <typename FieldTagsList>
+struct ErrorsCompute;
+
+template <typename... FieldTags>
+struct ErrorsCompute<tmpl::list<FieldTags...>>
+    : Errors<tmpl::list<FieldTags...>>, db::ComputeTag {
+  using field_tags = tmpl::list<FieldTags...>;
+  using base = Errors<tmpl::list<FieldTags...>>;
+  using return_type = typename base::type;
+  // using argument_tags =
+  //     tmpl::list<AnalyticSolutions<field_tags>,
+  //     ::Tags::Variables<field_tags>>;
```

Comment:
Why did you go with the other implementation? With this one, we could drop the subitems logic altogether, and also remove the single-tensor `std::optional` tags, right?